torch/testing/_internal/common_fsdp.py

```diff
@@ -1374,15 +1374,27 @@ class FullyShardPatch(Enum):
     def decorator(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
+            original_fully_shard = torch.distributed._composable.fsdp.fully_shard
```

Comment:
Is each thread ensured to read the original `torch.distributed._composable.fsdp.fully_shard` here? Is it that because the multiple threads are not spawned until `func(*args, **kwargs)` below?

Reply:
each thread patch ``test_fully_shard_training.fully_shard`` and no thread is patching ``torch.distributed._composable.fsdp.fully_shard``. so ``torch.distributed._composable.fsdp.fully_shard`` is the same to every thread